src/coreclr/vm/threads.cpp

```diff
@@ -1039,6 +1048,13 @@ void InitThreadManager()
     }
     CONTRACTL_END;
 
+#ifdef TARGET_WINDOWS
+    if (!InitFlsSlot())
+    {
+        _ASSERTE_ALL_BUILDS(!"Initialization of a FLS slot failed.");
```

Comment:
An exception would be more appropriate for this one. It would be even better if InitFlsSlot returns void and throws the exception right away.